tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -9740,99 +9749,141 @@ TEST_F(ExecutionTest, DynamicResourcesUniformIndexingTest) {
       std::make_shared<st::ShaderOpSet>();
   st::ParseShaderOpSetFromStream(pStream, ShaderOpSet.get());
   st::ShaderOp *pShaderOp =
-      ShaderOpSet->GetShaderOp("DynamicResourcesUniformIndexing");
+      ShaderOpSet->GetShaderOp("DynamicResourcesDynamicIndexing");
 
   bool Skipped = true;
 
   //D3D_SHADER_MODEL TestShaderModels[] = {D3D_SHADER_MODEL_6_0}; // FALLBACK
-  D3D_SHADER_MODEL TestShaderModels[] = {D3D_SHADER_MODEL_6_6};
+  D3D_SHADER_MODEL TestShaderModels[] = {D3D_SHADER_MODEL_6_0, D3D_SHADER_MODEL_6_6};
+
+  const int expectedResultsSize = 16;
+  float expectedResultsUniform[expectedResultsSize] = {
+    10.0, 10.0, 
+    12.0, 12.0,
+    14.0, 14.0, 
+    20.0, 20.0, 
+    22.0, 22.0,
+    24.0, 24.0, 
+    30.0, 30.0, 
+    32.0, 32.0};
+
+  float expectedResultsNonUniform[expectedResultsSize] = {
+    10.0, 11.0, 
+    12.0, 13.0,
+    14.0, 15.0, 
+    20.0, 21.0, 
+    22.0, 23.0,
+    24.0, 25.0, 
+    30.0, 31.0, 
+    32.0, 33.0};
+  float *expectedResultsArray[2];
+  expectedResultsArray[0] = expectedResultsUniform;
+  expectedResultsArray[1] = expectedResultsNonUniform;
+    
+  for (unsigned int non_uniform_bit = 0; non_uniform_bit < 2; non_uniform_bit++) {
+    for (unsigned i = 0; i < _countof(TestShaderModels); i++) {
```

Comment:
I think the nested order of these loops should be reversed, then device and feature detection moved outside the uniformity loop.